test.lua

```diff
@@ -584,6 +584,59 @@ for k, v in pairs(quoting_tests) do
     end
 end
 
+---- docstring tests ----
+
+print("Running tests for metadata and docstrings...")
+local docstring_tests = {
+    ['(doc (fn [] :A 1))'] = {'(_0_)\\n  A', 'no anonymous fn docstring mistmatch'},
+    ['(doc (λ [a] :B :ret-str))'] = {'(_0_ a)\\n  B', 'no anonymous lambda docstring mistmatch'},
+    -- TODO: set correct fnName based on assigned var/prop name and use following test
+    -- ['(local foo (fn [a] :C 1)) (doc foo)'] = {'(foo a)\\n  C', 'named/non-local fn docstring mismatch'},
+    ['(doc (fn foo [a] :C 1))'] = {'(foo a)\\n  C', 'no named fn docstring mismatch'},
+    ['(doc (λ foo [] :D 1))'] = {'(foo)\\n  D', 'no anonymous lambda docstring mistmatch'},
```

Comment:
Typo "mistmatch" in a few places. Really minor, but I hope you agree it is worth catching before this get merged.

@jaawerth BTW this is awesome and I'm really looking forward to seeing it merged. Thanks!

Reply:
Thanks! I actually got rid of that wording in the last update but I appreciate you calling it out.